crates/wysiwyg/src/composer_model/mentions.rs

```diff
@@ -77,7 +77,16 @@ where
         let (start, end) = self.safe_selection();
         let range = self.state.dom.find_range(start, end);
 
-        let new_node = DomNode::new_mention(url, text, attributes);
+        // use the display text or the presence of a `data-mention-type => at-room` attribute to decide the mention type
+        let new_node = if text == "@room".into()
```

Comment:
Eventually this logic could be shared with the HTML parsing logic (matrix-org/matrix-rich-text-editor#695 / matrix-org/matrix-rich-text-editor#692). So we could add a note here to remind us to update when it's ready.